x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/HasPrivilegesResponse.java

```diff
@@ -22,24 +25,35 @@
 /**
  * Response for a {@link HasPrivilegesRequest}
  */
-public class HasPrivilegesResponse extends ActionResponse {
+public class HasPrivilegesResponse extends ActionResponse implements ToXContentObject {
+    private String username;
     private boolean completeMatch;
     private Map<String, Boolean> cluster;
     private List<ResourcePrivileges> index;
     private Map<String, List<ResourcePrivileges>> application;
 
     public HasPrivilegesResponse() {
-        this(true, Collections.emptyMap(), Collections.emptyList(), Collections.emptyMap());
+        this("", true, Collections.emptyMap(), Collections.emptyList(), Collections.emptyMap());
     }
 
-    public HasPrivilegesResponse(boolean completeMatch, Map<String, Boolean> cluster, Collection<ResourcePrivileges> index,
+    public HasPrivilegesResponse(String username, boolean completeMatch, Map<String, Boolean> cluster, Collection<ResourcePrivileges> index,
                                  Map<String, Collection<ResourcePrivileges>> application) {
         super();
+        this.username = username;
         this.completeMatch = completeMatch;
         this.cluster = new HashMap<>(cluster);
-        this.index = new ArrayList<>(index);
+        this.index = sorted(new ArrayList<>(index));
```

Comment:
I would be using a `Set` in this circumstances.

Reply:
If I was starting from scratch, then so would I, but that would require changing the return types on `getIndexPrivileges` and `getApplicationPrivileges` which seemed more invasive than was justified by this PR.

I'm happy to revist that choice if you want - I went back-and-forth many times while making the change.

Reply:
It's all yours :)